packages/components/nodes/memory/AgentMemory/MySQLAgentMemory/mysqlSaver.ts

```diff
@@ -164,15 +187,44 @@ export class MySQLSaver extends BaseCheckpointSaver implements MemoryMethods {
         }
     }
 
+    private async cleanupOldMessages(dataSource: DataSource, threadId: string, limit: number = 100): Promise<void> {
```

Comment:
Can we have the option for user to enable/disable this. For example we can put this to the `this.inputs`:

```
{
                label: 'Auto Clear Messages',
                name: 'autoClearMessages',
                type: 'boolean',
                description: 'Automatically clear all but the last 100 messages for this thread',
                additionalParams: true,
                optional: true
            },
```